src/main/java/com/multithreading/App.java

```diff
@@ -1,11 +1,21 @@
 package com.multithreading;
 
-/**
- * Hello world!
- *
- */
+import java.util.Scanner;
+
 public class App {
     public static void main(String[] args) {
-        System.out.println("Hello World!");
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Enter a string: ");
+        String input = scanner.nextLine();
+        scanner.close();
+
+        TextProcessor processor = new TextProcessor();
+
+        try {
+            System.out.println("Capitalized String: " + processor.process(input));
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
that won't work though. You have N threads (N=string length) running, one was interrupted, the others are still running and you are only interrupting the main thread.

You  need to find a better way to end your threads.
One approach i recommend is to never interrupt your threads instead tell them to die with a volatile boolean. For example if you have threads that are writing to a file and you interrupt in the middle of it you'll end up with a corrupted file.